scripts/generate_changelog.py

```diff
@@ -0,0 +1,168 @@
+        return (None, [])
+
+
+def commit_title_pr_number(commit: Any) -> Tuple[str, Optional[int]]:
+    match = re.match(r"(.*) \(#(\d+)\)", commit.summary)
+    if match:
+        return (str(match.group(1)), int(match.group(2)))
+    else:
+        return (commit.summary, None)
+
+
+def print_section(title: str, items: List[str]) -> None:
+    if 0 < len(items):
+        print(f"#### {title}")
+        for line in items:
+            print(f"- {line}")
+    print()
+
+
+repo = Repo(".")
+commits = list(repo.iter_commits(COMMIT_RANGE))
+commits.reverse()  # Most recent last
+
+# Sections:
+analytics = []
+enhancement = []
+bugs = []
+dev_experience = []
+docs = []
+examples = []
+misc = []
+performance = []
+python = []
+renderer = []
+rfc = []
+rust = []
+ui = []
+viewer = []
+web = []
+
+for commit in tqdm(commits, desc="Processing commits"):
+    (title, pr_number) = commit_title_pr_number(commit)
+    if pr_number is None:
+        # Someone committed straight to main:
+        summary = f"{title} [{commit.hexsha}](https://github.com/{OWNER}/{REPO}/commit/{commit.hexsha})"
+        misc.append(summary)
+    else:
+        (pr_title, labels) = pr_title_labels(pr_number)
+        title = pr_title or title  # We prefer the PR title if available
+        summary = f"{title} [#{pr_number}](https://github.com/{OWNER}/{REPO}/pull/{pr_number})"
+
+        added = False
+
+        if labels == ["⛴ release"]:
+            # Ignore release PRs
+            continue
+
+        # Some PRs can show up underm multiple sections:
+        if "🐍 python API" in labels:
+            python.append(summary)
+            added = True
+        if "🦀 rust SDK" in labels:
+            rust.append(summary)
+            added = True
+        if "📊 analytics" in labels:
+            analytics.append(summary)
+            added = True
+
+        if not added:
+            # Put the remaining PRs under just one section:
+            if "🪳 bug" in labels or "💣 crash" in labels:
+                bugs.append(summary)
+            elif "📉 performance" in labels:
+                performance.append(summary)
+            elif "examples" in labels:
+                examples.append(summary)
+            elif "📖 documentation" in labels:
+                docs.append(summary)
+            elif "ui" in labels:
+                ui.append(summary)
+            elif "📺 re_viewer" in labels:
+                viewer.append(summary)
+            elif "🔺 re_renderer" in labels:
+                renderer.append(summary)
+            elif "🕸️ web" in labels:
+                web.append(summary)
+            elif "enhancement" in labels:
+                enhancement.append(summary)
+            elif "🧑‍💻 dev experience" in labels:
+                dev_experience.append(summary)
+            elif "💬 discussion" in labels:
+                rfc.append(summary)
+            elif not added:
+                misc.append(summary)
```

Comment:
I wonder if we should exclude `regression` tagged PRs 🤔 
depends obv. for how long a thing regressed

Reply:
You mean add a separate category for regression fixes?